From 323a85eb76b83d8880e0640d94ead9395a3a4472 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Sat, 19 Apr 2014 17:21:56 +0300
Subject: [PATCH] netlink: honor configured route-table for
 shortcut-destination ifaces

helpful if dynmic routing is used also internally and the routes
are placed on alternate table.
---
 nhrp/sysdep_netlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nhrp/sysdep_netlink.c b/nhrp/sysdep_netlink.c
index 1217ae4..44063ad 100644
--- a/nhrp/sysdep_netlink.c
+++ b/nhrp/sysdep_netlink.c
@@ -739,7 +739,7 @@ static void netlink_route_new(struct nlmsghdr *msg)
 
 	if (iface->flags & NHRP_INTERFACE_FLAG_SHORTCUT_DEST) {
 		/* Local shortcut target routes */
-		if (rtm->rtm_table != RT_TABLE_MAIN)
+		if (rtm->rtm_table != iface->route_table)
 			return;
 		type = NHRP_PEER_TYPE_LOCAL_ADDR;
 	} else if (iface->flags & NHRP_INTERFACE_FLAG_CONFIGURED) {
-- 
2.4.6

